### ISSUE TITLE
Error from gatsby-plugin-graphql-codegen v.2.1.4

### ISSUE THREAD

**Issue by joewood** [d4rekanguok/gatsby-typescript#37]
I'm seeing the following error when using the latest version:
```
results.filter(...).flat is not a function

  TypeError: results.filter(...).flat is not a function
  
  - graphql-codegen.config.js:88 
```

It looks like TypeScript is not transpiling `array.flat`. The reason for this is that `flat` has moved from `esnext` to `es2019`. If you change you lib array in **tsconfig** to include `es2019` it should fix it:

https://github.com/d4rekanguok/gatsby-typescript/blob/master/packages/gatsby-plugin-graphql-codegen/tsconfig.json#L6

Should be:
```json
  "lib": ["dom", "esnext", "es2019"], 
```

**Comment by d4rekanguok** [comment]
Whoa, I didn't realized that  array.flat wasn't being transpilied. Thank you very much for taking the time (if you feel like sending a PR I would gladly merge it) & sorry about the inconvenience!

**Comment by d4rekanguok** [comment]
TIL Typescript doesn't provide polyfill for new built-ins, so including ES2019 in --lib doesn't solve the problem. I've reverted back to using reduce so it'll run on nodejs < 11 (#38). Thank you again for taking the time & bringing this issue to light!

The fix is published in gatsby-plugin-graphql-codegen@2.2.1
